### Задание 0 (разминка)

Даны два списка `a` и `b` (см. ниже).
    
1. Выведите на экран первый элемент списка `a`, третий элемент списка `a`, последний элемент списка `a`.
2. Добавьте в список `b` элемент 7 (просто допишите в конец).
3. Замените пятый элемент списка `a` на число 8.
4. Создайте список `merged`, который включает в себя все элементы списка `a` и списка `b`.
5. Создайте новый список `с`, который получается заменой последнего элемента списка `a` на число 100. Сам список `a` измениться не должен!

In [ ]:
a = [1, 0, 9, 12, 18, 34, 89, 91, 33, 127]
b = [2, 8, 9, 11, 76, 25, 44, 7]
print(a[0], a[2], a[len(a)-1])

1 9 127


In [ ]:
a[4] = 8
print(a)

[1, 0, 9, 12, 8, 34, 89, 91, 33, 127]


In [ ]:
merged = a + b
print(merged)

[1, 0, 9, 12, 8, 34, 89, 91, 33, 127, 2, 8, 9, 11, 76, 25, 44, 7]


In [ ]:
c = a
c[-1] = 100
print(c)

[1, 0, 9, 12, 8, 34, 89, 91, 33, 100]


### Задание 1 (девочковое)

Есть список имен:

In [ ]:
girls = ["Иветта", "Виолетта", "Кассандра", "Вирджиния", 
         "Амелия", "Розамунда", "Янина", "Беатриса"]

Используя список `girls`, выведите на экран следующие списки:

    ["Виолетта", "Кассандра", "Вирджиния", "Амелия"]
    ["Вирджиния", "Амелия", "Розамунда", "Янина", "Беатриса"]
    ["Иветта", "Виолетта", "Вирджиния", "Амелия"]
    ["Кассандра", "Амелия", "Розамунда"]

In [ ]:
print(girls[1:5])

['Виолетта', 'Кассандра', 'Вирджиния', 'Амелия']


In [ ]:
print(girls[3:])

['Вирджиния', 'Амелия', 'Розамунда', 'Янина', 'Беатриса']


In [ ]:
girls[0:2]+girls[3:5]

['Иветта', 'Виолетта', 'Вирджиния', 'Амелия']

In [ ]:
girls[2:3]+girls[4:5]+girls[5:6]

['Кассандра', 'Амелия', 'Розамунда']

### Задание 2 (поэлементное) 

Дан список `L`. Выведите на экран (последовательно, с новой строчки): 

* все элементы списка `L`
* логарифмированные значения элементов списка `L`

In [ ]:
L = [12, 3, 8, 125, 10, 98, 54, 199]

In [ ]:
for i in L:
    print(i)

12
3
8
125
10
98
54
199


In [50]:
from math import log
L = [12, 3, 8, 125, 10, 98, 54, 199]
L2 = []
for i in L:
    L2.append(log (i)) 
for i in L2:
    print(i)

2.4849066497880004
1.0986122886681098
2.0794415416798357
4.8283137373023015
2.302585092994046
4.584967478670572
3.9889840465642745
5.293304824724492


In [ ]:
L = [12, 3, 8, 125, 10, 98, 54, 199]
L[4]=0
for i in L:
    print(i)

12
3
8
125
0
98
54
199


In [51]:
from math import log
L = [12, 3, 8, 125, 0, 98, 54, 199]
L2 = []
for i in L:
    L2.append(log (i)) 
for i in L2:
    print(i)

ValueError: ignored

Замените пятый элемент списка `L` на 0. Проделайте те же операции, что и раньше. Объясните, почему получаются такие результаты.

### Задание 3 (демографическое)

В списке `age` сохранены значения возраста респондентов: 

    age = [24, 35, 42, 27, 45, 48, 33]
    
Создайте список `age2`, в котором будут храниться значения возраста, возведенные в квадрат.

**Подсказка:** используйте цикл *for*.

In [ ]:
age = [24, 35, 42, 27, 45, 48, 33]
age1 = []

In [ ]:
for i in age:
    age1.append(i*i)
    print(i)

24
35
42
27
45
48
33


In [ ]:
age = [24, 35, 42, 27, 45, 48, 33]
age2 = []
for i in age:
    age2.append(i* i) 
print(age2)


[576, 1225, 1764, 729, 2025, 2304, 1089]


### Задание 4 (игровое)

Питон просит пользователя загадать (ввести с клавиатуры) целое число $k$ от 1 до 10. Питон берет это число $k$ и выводит на экран $k$-тый элемент списка `numbers`, причем $k$-ый элемент в привычном понимании, в нумерации, которая начинается с единицы.  

Список `numbers`:

In [ ]:
numbers = [1, 5, 6, 8, 10, 21, 25, 1, 0, -9, 9]

In [ ]:
n = int(input("Введите целое число от 1 до 10: "))
x=numbers[n-1]
print(x)

Введите целое число от 1 до 10: 3
6


**Пояснение:**
    
*Входные данные:*

    Введите целое число от 1 до 10: 3
    
*Выходные данные:*

    6

### Задание 5 (числовое)

У питона есть набор любимых чисел.

    favorites = [3, 7, 11, 23, 18, 48, 81]

Напишите программу, которая просит пользователя ввести целое число, и если оно нравится питону, на экран будет выводиться сообщение: "Мое любимое число!", если нет ‒ "Эх, ну почему?".

*Подсказка:* чтобы проверить, принадлежит ли некоторый объект элементам списка, можно воспользоваться оператором *in*.

In [ ]:
favorites = [3, 7, 11, 23, 18, 48, 81]
n = int(input("Введите целое число: "))
print("Мое любимое число!") if n in favorites else print("Эх, ну почему?")





Введите целое число: 7
Мое любимое число!


### Задание 6 (четное)

Напишите программу, которая просит пользователя ввести число и, если оно чётное, выводит на экран сообщение: "Это число чётное", если нечётное ‒ "Это число нечётное".

*Подсказка:* используйте оператор `%` для определения остатка от деления.

In [ ]:
n = int(input("Введите целое число: "))
print("Это число чётное") if n % 2 == 0 else print("Это число нечётное")

Введите целое число: 1
Это число нечётное


### Задание 7 (музыкальное)

Напишите программу, которая принимает на вход список слов такого вида:

    words = ["Speak ","to", "me ", "of", "Florence" ,"And ", "of", "the", "Renaissance"]

а возвращает список

    words_clean = ["speak", "to", "me", "of", "florence", "and", "of", "the", "renaissance"]

Другими словами, программа убирает пробелы в словах и приводит все слова к нижнему регистру. Эту задачу можно решить через циклы, но в Python есть такая полезная вещь как списковые включения. Они работают быстрее, чем циклы, и создавать новые списки на основе старых очень удобно. Прочитать про них можно [здесь](http://pythontutor.ru/lessons/lists/#section_3) и [здесь](https://nbviewer.jupyter.org/github/allatambov/Py-programming-3/blob/master/20-04/lecture-lists-contd.ipynb).

**Подсказка:** запросите `help()`  по  методам `.strip()` и `.lower()`.

In [49]:
words = input("Введите нужные слова через запятую: ")
word = words.split(',')
worda = [n.strip(" ") for n in word]
word_clean = [n.lower() for n in worda]
word_clean

Введите нужные слова через запятую: Мама, Папа   ,   Брат


['мама', 'папа', 'брат']

### Задание 8 (секретное)

Напишите программу, которая запрашивает у пользователя пароль, и далее:

- если пароль верный, выводит на экран сообщение "Login success".
- если пароль неверный, выводит на экран сообщение "Incorrect password, try again!" до тех пор, пока пользователь не введет верный пароль.

Считайте, что верный пароль программе известен, то есть уже сохранен в некоторой переменной. 

In [ ]:
x = 000000
n = int(input("Введите пароль: "))
while n != x:
  print("Incorrect password, try again!")
  n = int(input("Введите пароль: "))
else:
    print("Login success")

Введите пароль: 111256
Incorrect password, try again!
Введите пароль: 000004
Incorrect password, try again!
Введите пароль: 000000
Login success


### Задание 9 (функциональное)

Напишите функцию `nums()`, которая принимает на вход целое число, а возвращает список из двух элементов: целое число, предшествующее принятому на вход, и число, следующее за ним.

**Пример:**

*Входные данные:*

    7

*Выходные данные:*

    [6, 8]

In [ ]:
n = int(input("Введите целое число: "))
a = n-1
b = n+1
c = [a, b]
print(c)

Введите целое число: 18
[17, 19]


### Задание 10 (счетное)

Напишите функцию `my_log()`, которая принимает на вход список чисел, и возвращает список их натуральных логарифмов. Если число меньше или равно 0, на его месте в возвращаемом списке должно быть None.

**Пример:**

*Входные данные:*

    [1, 3, 2.5, -1, 9, 0, 2.71]

*Выходные данные:*

    [0.0,
     1.0986122886681098,
     0.9162907318741551,
     None,
     2.1972245773362196,
     None,
     0.9969486348916096]

In [57]:
numbers = input("Введите числа: ")
number = numbers.split()
numbera = []

for n in number:
  numbera.append(float(n))
numbera



Введите числа: 5 4.2 1 0 -5 -7


[5.0, 4.2, 1.0, 0.0, -5.0, -7.0]

In [88]:
from math import log
L = numbera
L1 = []
for i in L:
    L1.append(log(i)) if i > 0 else L1.append('NONE')
for i in L1:
    print(i)


1.6094379124341003
1.4350845252893227
0.0
NONE
NONE
NONE


### Задание 11 (зимнее)

Напишите функцию `str_lower()`, которая принимает на вход строку (набор слов через пробел), а возвращает список ее элементов в нижнем регистре. 

**Пример:**

*Входные данные:*

    "В лесу родилась ёлочка В лесу она росла"

*Выходные данные:*

    ['в', 'лесу', 'родилась', 'ёлочка', 'в', 'лесу', 'она', 'росла']
    
*Подсказка:* для разбиение строки на части, познакомьтесь с методом `.split()` – запросите по нему `help()`.

In [41]:
words = input("Введите слова: ")
word = words.split()
word_clean = [n.lower() for n in word]
word_clean

Введите слова: В лесу родилась ёлочка


['в', 'лесу', 'родилась', 'ёлочка']